### PR TITLE
Reload apps list after create app

### PR DESCRIPTION
**Pull request by StephenAraka**
### What does this PR do?
- automatically reload apps list after creating an app

### Where the help is needed...
After creating an app, the state in redux is changed... however, this change does not reflect in the component.
I tried using await but no good. I'll need help with this. Maybe we can have a call with someone and I can clarify more.

### CODE REVIEW

**Comment by StephenAraka** [comment]
So, I've rebased develop-new into this branch...

**Comment by StephenAraka** [comment]
> Great work @StephenAraka , the page loads immediately the app is created, which is fine. Issue is `popup Model` would close on successful `App` creation, now it doesn't close, nor does it give feedback to the `user`. Do you mind taking care of that issue

@Mubangizi I realized this issue as well... The modal closes on 2nd click, same thing happens with delete app... But should it be fixed on this PR, or another?

**Comment by StephenAraka** [comment]
> Great work @StephenAraka , the page loads immediately the app is created, which is fine. Issue is `popup Model` would close on successful `App` creation, now it doesn't close, nor does it give feedback to the `user`. Do you mind taking care of that issue

let me fix this first

File: src/components/AppsList/index.jsx [hunk 1]
```diff
@@ -65,7 +72,12 @@ class AppsList extends Component {
 AppsList.propTypes = {
   apps: PropTypes.arrayOf(PropTypes.object),
   isRetrieved: PropTypes.bool,
```
Comment (ckwagaba) [inline src/components/AppsList/index.jsx:74]: please explain these two props: `isRetrieving` and `isRetrieved`

Comment (StephenAraka) [reply src/components/AppsList/index.jsx:74]: Well @ckwagaba:
- `isRetrieving` - is a boolean loading state (used to display Spinner) 
- `isRetrieved` - is another boolean state (used to indicate whether the loading is complete and successful)

@Mubangizi did this... he can confirm if I'm correct.

Comment (Mubangizi) [reply src/components/AppsList/index.jsx:74]: Yeah, we use them for `spinner` and for `on failure` feedback, thats if is retrieved stays false

**Comment by Mubangizi** [comment]
> > Great work @StephenAraka , the page loads immediately the app is created, which is fine. Issue is `popup Model` would close on successful `App` creation, now it doesn't close, nor does it give feedback to the `user`. Do you mind taking care of that issue
> 
> @Mubangizi I realized this issue as well... The modal closes on 2nd click, same thing happens with delete app... But should it be fixed on this PR, or another?

Yeah, i guess you can do it here. because i thought `Create Apps` was like closing on `Success`, and that issue has been introduced now, in this PR. if thats the case then you could take time and finish it, once and for all what do you say

**Comment by Mubangizi** [comment]
Which help do you want BTW? you can put the issue in the PR description

**Comment by StephenAraka** [comment]
> Which help do you want BTW? you can put the issue in the PR description

Added the issue... Please check out the description

**Comment by Mubangizi** [comment]
> > Which help do you want BTW? you can put the issue in the PR description
> 
> Added the issue... Please check out the description

Yeah sure, that exact thing was happening to me before when i was using `componentDidUpdate`. I dont know why also tried the `waiting` but it seemed like we always called to component so early.
Although, which is not helpful now, when i tried `window.location.reload` it would bring backt the right results.

**Pull request merged**